### DISCUSSION TITLE
Segment creation and what it should output

### DISCUSSION THREAD

**Discussion by estruyf** [JanDeDobbeleer/oh-my-posh#1891] [Troubleshoot] [answered]
Hi all,

I just started creating my first segment and like this, all is a bit new to me, I hope you could help me out. Using the code sample provided in the docs (https://ohmyposh.dev/docs/contributing_segment), there is a `{{.Text}}` variable defined, but how can you pass a value?

My configuration looks as follows:

```json
{
  "type": "npm",
  "style": "plain",
  "foreground": "#193549",
  "background": "#ffeb3b",
  "properties": {
    "newprop": "Hello"
  }
}
```

This is what it renders:

![Screenshot 2022-03-10 at 17 26 39](https://user-images.githubusercontent.com/2900833/157708907-be9f7f4a-d2d3-4e1b-ac10-1f3ef6144403.png)

**Questions**

1. Should the output be: `hello world`?
2. I would expect that the `newprop` needs to be defined somewhere in the template. Is there some magic behind the `{{.Text}}` happening?

**Comment by estruyf** [comment, marked as answer]
Or should the segment `Init` method be:

```go
func (n *Npm) Init(props properties.Properties, env environment.Environment) {
    n.props = props
    n.env = env

	n.Text = props.GetString(NewProp, "Hello")
}
```

**Reply by JanDeDobbeleer** [reply]
You're right, that seems to be missing 😅
